src/index.d.ts

```diff
@@ -115,6 +139,7 @@ declare module 'bootstrap-vue' {
   export interface Progress extends Vue {}
   export interface Table extends Vue {
     refresh: () => void
+    fields?: TableFieldObject | TableFieldArray
```

Comment:
Should `fields` be up in `export interface TableField {`? and we are missing `items`, which could be `array[object]` | function`, where function is a provider function?

Reply:
Not to jump in but, the first request doesn't make sense: `fields` is a property on a table is it not? Not a property on the "field" itself.

In general, I don't see the use case for having anything other than methods that might be called externally via a reference. In 99% of uses you're going to be managing your fields / items (or provider method) passed into this component, and shouldn't need to reference them from the table - they should instead be modified/referenced via whatever is consuming the `b-table` component.
Ex:
```html
<b-table :items="myItems" :fields="myFields" ref="table1">....</b-table>
```
The code that this is in will have `myItems` and `myFields`... in what cases would I need to do:
`(this.$refs.table1 as Table).fields` or `(this.$refs.table1 as Table).items`? Other than reading them, modifying them goes against Vue conventions and recommended practices

Reply:
Ah, right... I was thinking it was the table interface definition, not the fields definition (jumping back and forth between too many things at the moment)